tests/conftest.py

```diff
@@ -149,6 +149,15 @@ def pytest_configure(config: pytest.Config) -> None:
     if config.getoption("verbose") > 0:
         logging.getLogger().setLevel(logging.DEBUG)
 
+    # pylint: disable-next=import-outside-toplevel
+    from syrupy.session import SnapshotSession
+
+    # pylint: disable-next=import-outside-toplevel
+    from .syrupy import override_syrupy_finish
+
+    # Override default finish to detect unused snapshots despite xdist
+    SnapshotSession.finish = override_syrupy_finish
```

Comment:
This sounds like a workaround for an upstream issue?

Reply:
> This sounds like a workaround for an upstream issue?

Yes it is a workaround for an upstream issue.
There is a PR open there, but it fills a gap right now, see https://github.com/home-assistant/core/pull/128477

> I honestly really don't think this belongs in our codebase.

I have added a comment to indicate that it can be removed when upstream is implemented